packages/next/src/server/base-server.ts

```diff
@@ -1315,6 +1315,11 @@ export default abstract class Server<ServerOptions extends Options = Options> {
       opts.isBot = isBotRequest
     }
 
+    // In development, we always want to generate dynamic HTML.
+    if (opts.dev && opts.supportsDynamicHTML === false) {
```

Comment:
```suggestion
    if (isAppDir && opts.dev && opts.supportsDynamicHTML === false) {
```
Should be specific to app paths.